pkg/testcase/cluster.go

```diff
@@ -11,39 +11,105 @@ import (
 	. "github.com/onsi/gomega"
 )
 
+var arch string
+
 func TestBuildCluster(g GinkgoTInterface) {
+	var err error
+
 	cluster := factory.GetCluster(g)
+	arch = cluster.ArchType
 	Expect(cluster.Status).To(Equal("cluster created"))
+	Expect(shared.KubeConfigFile).ShouldNot(BeEmpty())
+	Expect(cluster.ServerIPs).ShouldNot(BeEmpty())
 
-	if strings.Contains(cluster.ClusterType, "etcd") {
-		fmt.Println("Backend:", cluster.ClusterType)
-	} else {
-		fmt.Println("Backend:", cluster.ExternalDb)
+	if cluster.ProductType == "k3s" {
+		if strings.Contains(cluster.ClusterType, "etcd") {
+			fmt.Println("Backend:", cluster.ClusterType)
+		} else {
+			fmt.Println("Backend:", cluster.ExternalDb)
+		}
+	
+		if cluster.ExternalDb != "" && cluster.ClusterType == "" {
+			for i := 0; i > len(cluster.ServerIPs); i++ {
+				cmd := "grep \"datastore-endpoint\" /etc/systemd/system/k3s.service"
+				res, err := shared.RunCommandOnNode(cmd, cluster.ServerIPs[0])
+				Expect(err).NotTo(HaveOccurred())
+				Expect(res).Should(ContainSubstring(cluster.RenderedTemplate))
+			}
+		}
 	}
 
-	if cluster.ExternalDb != "" && cluster.ClusterType == "" {
-		for i := 0; i > len(cluster.ServerIPs); i++ {
-			cmd := "grep \"datastore-endpoint\" /etc/systemd/system/k3s.service"
-			res, err := shared.RunCommandOnNode(cmd, cluster.ServerIPs[0])
-			Expect(err).NotTo(HaveOccurred())
-			Expect(res).Should(ContainSubstring(cluster.RenderedTemplate))
-		}
+	fmt.Println("\nKUBECONFIG:\n")
+	err = shared.PrintFileContents(shared.KubeConfigFile)
+	if err != nil {
+		return
 	}
 
-	err := shared.PrintFileContents(shared.KubeConfigFile)
+	fmt.Println("BASE64 ENCODED KUBECONFIG:\n")
+	err = shared.PrintBase64Encoded(shared.KubeConfigFile)
 	if err != nil {
 		return
 	}
 
-	Expect(shared.KubeConfigFile).ShouldNot(BeEmpty())
-	Expect(cluster.ServerIPs).ShouldNot(BeEmpty())
+	fmt.Println("\nServer Node IPS:", cluster.ServerIPs)
+
+	checkAndPrintAgentNodeIPs(cluster.NumAgents, cluster.AgentIPs, false)
+
+	if cluster.ProductType == "rke2" {
+		checkAndPrintAgentNodeIPs(cluster.NumWinAgents, cluster.WinAgentIPs, true)
+	}	
+}
+
+// TestSonobuoyMixedOS runs sonobuoy tests for mixed os cluster (linux + windows) node
+func TestSonobuoyMixedOS(version string, delete bool) {
+	err := shared.SonobuoyMixedOS("install", version)
+	if err != nil {
+		fmt.Println(err)
+		return
```

Comment:
on those returns on tests , since all of our test functions do not return errors , in return statements that we need to break the flow because of an error like here, you can do something like :

```

if err != nil {
GinkgoT().Errorf("error {your custom message if needed }: %v", err)
}
```

Reply:
Updated